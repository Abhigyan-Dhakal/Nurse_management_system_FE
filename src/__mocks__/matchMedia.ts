export default Object.defineProperty(window, "matchMedia", {
  value: () => {
    return {
      matches: false,
      addListener: () => {},
      removeListener: () => {},
    };
  },
});
