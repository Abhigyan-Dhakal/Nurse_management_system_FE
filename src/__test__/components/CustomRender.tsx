import type { PreloadedState } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { render } from "@testing-library/react";
import React, { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { allReducers, RootState, store } from "../../redux/store";
export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: allReducers,
    preloadedState,
  });
};
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
}
export function renderWithProvidersAndNavigation(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
