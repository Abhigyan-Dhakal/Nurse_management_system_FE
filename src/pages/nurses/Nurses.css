.nurses-container {
  padding: 40px 80px;
}

.nurses-heading {
  font-size: 42px;
  color: rgb(29, 84, 156);
}
