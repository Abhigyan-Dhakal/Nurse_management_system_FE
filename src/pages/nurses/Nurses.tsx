import { Divider } from "antd";
import { NurseTable } from "../../components";

import "./Nurses.css";

type Props = {};
export const Nurses = (props: Props) => {
  return (
    <div className="nurses-container">
      <div>
        <h1 className="nurses-heading">All Nurses</h1>
      </div>
      <Divider />
      <div>
        <NurseTable />
      </div>
    </div>
  );
};
