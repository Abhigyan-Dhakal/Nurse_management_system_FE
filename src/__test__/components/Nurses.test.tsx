import "../../__mocks__/matchMedia";
import { Nurses } from "../../pages/nurses/Nurses";
import { renderWithProvidersAndNavigation } from "./CustomRender";

export const nurseArray = [
  {
    nurse_id: 1,
    name: "Test Nurse",
    email: "nurseTest@gmail.com",
    photograph: "PhotoTest",
    workingDays: 12,
    dutyStartTime: "10AM",
    dutyEndTime: "5PM",
    isRoundingManager: true,
    address: "KamalPokhari",
    contact: "1234567890",
  },
];

it("should render nurse screen properly", () => {
  const page = renderWithProvidersAndNavigation(<Nurses />, {
    preloadedState: {
      nurseReducer: { data: nurseArray },
    },
  });
  expect(page.asFragment()).toMatchSnapshot();
});
