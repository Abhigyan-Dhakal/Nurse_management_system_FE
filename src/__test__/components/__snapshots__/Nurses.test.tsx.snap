// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render nurse screen properly 1`] = `
<DocumentFragment>
  <div
    class="nurses-container"
  >
    <div>
      <h1
        class="nurses-heading"
      >
        All Nurses
      </h1>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div>
      <div>
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            Rounding Manager
                          </th>
                          <th
                            class="ant-table-cell"
                          />
                          <th
                            class="ant-table-cell"
                          >
                            Name
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Email
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Working Days
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Duty Start Time
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Duty End Time
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Address
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Contact
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Action
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="favourite-container"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                                style="color: orange;"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <img
                              alt="PhotoTest"
                              class="profile-img"
                              src="PhotoTest"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Test Nurse
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            nurseTest@gmail.com
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            12
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Invalid date
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Invalid date
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            KamalPokhari
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            1234567890
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space ant-space-horizontal ant-space-align-center"
                            >
                              <div
                                class="ant-space-item"
                                style="margin-right: 8px;"
                              >
                                <a
                                  href="/1/edit-nurse"
                                >
                                  <div
                                    class="edit-link"
                                  >
                                    Edit
                                  </div>
                                </a>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <div
                                  class="delete-link"
                                >
                                  Delete
                                </div>
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
